docs/source/hooks/hooks.md

```diff
@@ -0,0 +1,139 @@
+* `after_dataset_loaded`
+* `before_dataset_saved`
+* `after_dataset_saved`
+* `after_context_created`
+
+The naming convention for non-error Hooks is `<before/after>_<noun>_<past_participle>`, in which:
+
+* `<before/after>` and `<past_participle>` refers to when the Hook executed, e.g. `before <something> was run` or `after <something> was created`.
+* `<noun>` refers to the relevant component in the Kedro execution timeline for which this Hook adds extra behaviour, e.g. `catalog`, `node` and `pipeline`.
+
+The naming convention for error hooks is `on_<noun>_error`, in which:
+
+* `<noun>` refers to the relevant component in the Kedro execution timeline that throws the error.
+
+[kedro.framework.hooks](/kedro.framework.hooks) lists the full specifications for which you can inject additional behaviours by providing an implementation.
+
+
+#### CLI hooks
+
+Lastly, Kedro defines a small set of CLI hooks that inject additional behaviour around execution of a Kedro CLI command:
+
+* `before_command_run`
+* `after_command_run`
+
+This is what the [`kedro-telemetry` plugin](https://github.com/kedro-org/kedro-plugins/tree/main/kedro-telemetry) relies on under the hood in order to be able to collect CLI usage statistics.
+
+### Hook implementation
+
+You should provide an implementation for the specification that describes the point at which you want to inject additional behaviour. The Hook implementation should have the same name as the specification. The Hook must provide a concrete implementation with a subset of the corresponding specification's parameters (you do not need to use them all).
+
+To declare a Hook implementation, use the `@hook_impl` decorator.
+
+For example, the full signature of the [`after_data_catalog_created`](/kedro.framework.hooks.specs.DataCatalogSpecs) Hook specification is:
+
+```python
+@hook_spec
+def after_catalog_created(
+    self,
+    catalog: DataCatalog,
+    conf_catalog: Dict[str, Any],
+    conf_creds: Dict[str, Any],
+    save_version: str,
+    load_versions: Dict[str, str],
+) -> None:
+    pass
+```
+
+However, if you just want to use this Hook to list the contents of a data catalog after it is created, your Hook implementation can be as simple as:
+
+```python
+# <your_project>/src/<your_project>/hooks.py
+import logging
+
+from kedro.framework.hooks import hook_impl
+from kedro.io import DataCatalog
+
+
+class DataCatalogHooks:
+    @property
+    def _logger(self):
+        return logging.getLogger(self.__class__.__name__)
+
+    @hook_impl
+    def after_catalog_created(self, catalog: DataCatalog) -> None:
+        self._logger.info(catalog.list())
+```
+
+```{note}
+The name of a module that contains Hooks implementation is arbitrary and is not restricted to `hooks.py`.
+```
+
+We recommend that you group related Hook implementations under a namespace, preferably a class, within a `hooks.py` file that you create in your project.
+
+#### Registering your Hook implementations with Kedro
```

Comment:
- #### [Registering your Hook implementations with Kedro](https://github.com/kedro-org/kedro/edit/main/docs/source/extend_kedro/regHookImp.md)
- #### [Disable auto-registered plugins' Hooks](https://github.com/kedro-org/kedro/edit/main/docs/source/extend_kedro/disableHooks.md)

Again, splitting into separate files.

Reply:
Aren't these subheadings of `Hook implementation` would they make sense out of context on a separate page?

Reply:
I think when viewed together it makes sense the way this is organised. I'm not sure the information architecture would make sense if the sections were split out, it makes slightly more work for a reader to understand all of the Hooks concepts.